cmds/webboot/utils.go

```diff
@@ -66,3 +71,28 @@ func download(URL, fPath string) error {
 	verbose("%q is downloaded at %q\n", URL, fPath)
 	return nil
 }
+
+// getCachedDirectory recognizes the usb stick that contains the cached directory from block devices,
+// and return it's mount point.
+func getCachedDirectory() (*mount.MountPoint, error) {
+	blockDevs, err := block.GetBlockDevices()
+	if err != nil {
+		return nil, fmt.Errorf("No available block devices to boot from")
+	}
+
+	mountPoints, err := ioutil.TempDir("", "temp-device-")
+	if err != nil {
+		return nil, fmt.Errorf("Cannot create tmpdir: %v", err)
+	}
+
+	for _, device := range blockDevs {
+		mp, err := mount.TryMount(filepath.Join("/dev/", device.Name), filepath.Join(mountPoints, device.Name), "", mount.ReadOnly)
+		if err != nil {
+			continue
+		}
+		if _, err = os.Stat(filepath.Join(mp.Path, "Image")); err == nil {
+			return mp, nil
+		}
+	}
+	return nil, fmt.Errorf("Do not find the cache directory")
```

Comment:
I think in this error message you'd similarly want to explain to the user what was expected - you expected to find a directory named Images/ at the root of the block device right?